src/Ev.ServiceBus.Abstractions/Configuration/QueueOptions.cs

```diff
@@ -1,29 +1,36 @@
 ﻿using System;
 using Microsoft.Azure.ServiceBus;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 
 // ReSharper disable once CheckNamespace
 namespace Ev.ServiceBus.Abstractions
 {
     public class QueueOptions : ReceiverOptions
     {
-        public QueueOptions(string queueName) : base(queueName)
+        private readonly IServiceCollection _serviceCollection;
+
+        public QueueOptions(IServiceCollection serviceCollection, string queueName) : base(queueName, ClientType.Queue)
         {
+            _serviceCollection = serviceCollection;
             QueueName = queueName;
         }
 
         public string QueueName { get; }
 
-        public QueueOptions WithCustomMessageHandler<TMessageHandler>(Action<MessageHandlerOptions> config = null)
-            where TMessageHandler : IMessageHandler
+        public QueueOptions WithCustomMessageHandler<TMessageHandler>(Action<MessageHandlerOptions>? config = null)
+            where TMessageHandler : class, IMessageHandler
         {
+            _serviceCollection.TryAddScoped<TMessageHandler>();
```

Comment:
Extend possibility to select the lifetime by the user?
Maybe there are scenarios when someone wants to add it as transient or singleton.

Reply:
Or do we assume, that someone will register it beforehand and that's why the "try" is here?

Reply:
The "Try" is there because you can register 2 queues/subscriptions to listen to the same handler. That's what actually happens in Ev.Protocols.IntegrationEvents and Ev.Cqrs.ServiceBusConnector.

Reply:
I don't give the choice to the user of the API on purpose because I don't see any value added into giving that choice.